torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -759,6 +759,11 @@ def propagate_meta_info(new_hop_gm, new_call_function_node, old_call_function_no
                 ),
             )
             propagate_meta_info(new_bw_hop_gm, new_bw_node, bw_node)
+            # Since the partitioner is run after the graph passes, we have lost
+            # the eager information and cannot faithfully extract the eager
+            # inputs for the new partitioned backward graph. For the forward
+            # graph, it was fine because the input signature remains same.
+            new_bw_node.meta.pop("eager_input_vals", None)
```

Comment:
er come again?

Reply:
The problem is that we save vals only for inputs, but after partitioning we have intermediate tensors stashed as outputs. We don't have the eager vals for the stashed tensors (because they were not inputs).

Second, the added line is only for min-cut partitioner on the HOPs. During the autograd key, we get the bw graph as the full fw+bw graph, and there we save the eager vals for the primals and tangents. During the min-cut, we decide the new stashed tensors and we dont have eager_input_vals for them. Therefore, the old `eager_input_vals` of the node is no longer valid.

Reply:
Are you just saying that the backward HOPs shouldn't get eager_input_vals but the forward HOPs should?

Reply:
Oh I see the problem, the partitioned inputs are different and we didn't save the eager_input_vals for those